rpc/amqp-rpc.go

```diff
@@ -147,11 +147,47 @@ func amqpSubscribe(ch *amqp.Channel, name string, consumerName string, log *blog
 		AmqpNoWait,
 		nil)
 	if err != nil {
-		log.Crit(fmt.Sprintf("Could not subscribe to queue: %s", err))
-		return nil, err
+		return nil, fmt.Errorf("Could not subscribe to queue: %s", err)
 	}
 
-	return msgs, err
+	return msgs, nil
+}
+
+// connect attempts to connect to a channel and subscribe to the named queue,
+// returning error if it fails. This is used at first startup, where we want to
+// fail fast if we can't connect.
+func connect(config cmd.Config, queueName string) (msgs <-chan amqp.Delivery, channel *amqp.Channel, closeChan chan *amqp.Error, err error) {
```

Comment:
please remove these named return values. They cause more confusion as functions grow than they are worth when the functions are small.

Also, given how many objects are being returned, it seems like there should be something that wraps them?

I think doing so would also give us a place to put the locking instead of having it littered in a few places. Could be wrong.